packages/examples/cvat/exchange-oracle/src/services/cvat.py

```diff
@@ -207,23 +207,50 @@ def update_project_statuses_by_escrow_address(
     escrow_address: str,
     chain_id: int,
     status: ProjectStatuses,
+    tasks_status: TaskStatuses | None = None,
+    jobs_status: JobStatuses | None = None,
 ) -> None:
-    statement = (
+    # Update the status of the projects and return the updated project IDs
+    project_update_statement = (
         update(Project)
         .where(
             Project.escrow_address == escrow_address,
             Project.chain_id == chain_id,
         )
         .values(status=status.value)
+        .returning(Project.cvat_id)
     )
-    session.execute(statement)
+    updated_project_ids = set(session.execute(project_update_statement).scalars())
+
+    if not updated_project_ids:
+        return
+
+    if tasks_status is not None:
+        task_update_statement = (
+            update(Task)
+            .where(Task.cvat_project_id.in_(updated_project_ids))
+            .values(status=tasks_status.value)
+        )
+        session.execute(task_update_statement)
+
+    if jobs_status is not None:
+        job_update_statement = (
+            update(Job)
+            .where(Job.cvat_project_id.in_(updated_project_ids))
+            .values(status=jobs_status.value)
+        )
+        session.execute(job_update_statement)
```

Comment:
I'm not sure it should be a part of the same function with such a name. Consider splitting it into 3 functions or adding one which deletes an escrow.

Reply:
After some consideration I've decided that there's indeed no need to change Job/Task status and reverted these changes.